pkg/target/baseandoverlaysmall_test.go

```diff
@@ -180,3 +180,62 @@ spec:
         name: whatever
 `)
 }
+
+func TestSmallOverlayJSONPatch(t *testing.T) {
+	th := NewKustTestHarness(t, "/app/overlay")
+	writeSmallBase(th)
+	th.writeK("/app/overlay", `
+bases:
+- ../base
+patchesJson6902:
+- target:
+    version: v1
+    kind: Service
+    name: myService # BUG: this should be a-myService, because that is what the output for the base contains
+  path: service-patch.yaml
```

Comment:
Not sure if I fully understand, the test fails if `name` is myService, and you expect `name` to be set to a-myService?

Reply:
The test passes as it is written now. But I consider that a bug (see https://github.com/kubernetes-sigs/kustomize/issues/972) and would expect the test to pass with `name: a-myService` instead.

Reply:
This is _great_.  Can you add a reference to 972 in the comment, or better yet in the name of the test?